types/joi/joi-tests.ts

```diff
@@ -988,13 +988,13 @@ const Joi3 = Joi.extend({
         {
             name: 'asd',
             params: {
-                allowF: Joi.boolean().default(false),
+                allowFalse: Joi.boolean().default(false),
             },
             setup(params) {
-                const fIsAllowed = params.allowF;
+                const fIsAllowed = params.allowFalse;
             },
-            validate(params, value, state, options) {
-                if (value === 'asd' || params.allowF && value === 'asdf') {
+            validate(params, value: boolean, state, options) {
```

Comment:
We might want "value" to be "any" here. In theory, you can validate any value against a schema, and we should only assert that the value is the expected type once validation passes.